recipes/libx264/all/conanfile.py

```diff
@@ -51,136 +45,125 @@ def config_options(self):
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
-        del self.settings.compiler.libcxx
-        del self.settings.compiler.cppstd
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.libcxx")
+        self.settings.rm_safe("compiler.cppstd")
 
     @property
     def _with_nasm(self):
         return self.settings.arch in ("x86", "x86_64")
 
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
     def build_requirements(self):
         if self._with_nasm:
-            self.build_requires("nasm/2.15.05")
-        if self._settings_build.os == "Windows" and not tools.get_env("CONAN_BASH_PATH"):
-            self.build_requires("msys2/cci.latest")
+            self.tool_requires("nasm/2.15.05")
+        if self._settings_build.os == "Windows":
+            self.win_bash = True
+            if not self.conf.get("tools.microsoft.bash:path", check_type=str):
+                self.tool_requires("msys2/cci.latest")
 
     def source(self):
-        get(self, **self.conan_data["sources"][self.version],
-            destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
-    @contextlib.contextmanager
-    def _build_context(self):
-        with tools.vcvars(self) if self._is_msvc else tools.no_op():
-            yield
+    def generate(self):
+        env = VirtualBuildEnv(self)
+        env.generate()
+
+        tc = AutotoolsToolchain(self)
 
-    @property
-    def env(self):
-        ret = super(LibX264Conan, self).env
-        ret.update(self._override_env)
-        return ret
-
-    def _configure_autotools(self):
-        if self._autotools:
-            return self._autotools
-        self._autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
-        self._autotools.libs = []
         extra_asflags = []
         extra_cflags = []
         extra_ldflags = []
-        args = [
-            "--bit-depth=%s" % str(self.options.bit_depth),
-            "--disable-cli",
-            "--prefix={}".format(tools.unix_path(self.package_folder)),
-        ]
+        args = {
+            "--bit-depth": self.options.bit_depth,
+            "--disable-cli": "",
+            "--sbindir": None,          # Not understood by configure
+            "--oldincludedir": None     # Not understood by configure
+        }
         if self.options.shared:
-            args.append("--enable-shared")
+            args["--enable-shared"] = ""
+            args["--disable-static"] = None # --disable-static is not understood
         else:
-            args.append("--enable-static")
+            args["--enable-static"] = ""
+            args["--disable-shared"] = None # --disable-shared is not understood
```

Comment:
--disable-static can be moved outside this branching I think, otherwise it' a little bit confusing I think.